components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -249,6 +250,11 @@ public Response apisPost(APIDTO body,String contentType){
                 RestApiUtil.handleBadRequest(
                         "Specified tier(s) " + Arrays.toString(invalidTiers.toArray()) + " are invalid", log);
             }
+            APIPolicy apiPolicy = apiProvider.getAPIPolicy(username, body.getApiLevelPolicy());
+            if (apiPolicy == null && body.getApiLevelPolicy() != null) {
```

Comment:
should we check  body.getApiLevelPolicy() as it's pass for previous statement I dont think we need to check it

Reply:
This is for if we ignore apiLevelPolicy parameter when invoking the API or set null for that it will be resource level policy, if I don't check null here, if we receive a a null for that parameter getAPIPolicy will return a null so apiPolicy will be null and it will go inside the if condition and through an exception.

Reply:
Ok great. Don't we have a validation in service layer?

Reply:
AFAIK we don't have